Mage.Sets/src/mage/cards/s/SwoopingPteranodon.java

```diff
@@ -0,0 +1,95 @@
+package mage.cards.s;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.Mode;
+import mage.abilities.common.EntersBattlefieldThisOrAnotherTriggeredAbility;
+import mage.abilities.common.delayed.AtTheBeginOfNextEndStepDelayedTriggeredAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DamageFromOneToAnotherTargetEffect;
+import mage.abilities.effects.common.UntapTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.effects.common.continuous.GainControlTargetEffect;
+import mage.constants.*;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.mageobject.AbilityPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class SwoopingPteranodon extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent(SubType.DINOSAUR, "Dinosaur with flying");
+
+    static {
+        filter.add(new AbilityPredicate(FlyingAbility.class));
+    }
+
+    public SwoopingPteranodon(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{R}{W}");
+        
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Haste
+        this.addAbility(HasteAbility.getInstance());
+
+        // Whenever Swooping Pteranodon or another Dinosaur with flying enters the battlefield under your control,
+        // gain control of target creature an opponent controls until end of turn. Untap that creature.
+        // It gains flying and haste until end of turn.
+        // At the beginning of the next end step, target land deals 3 damage to that creature.
+        Ability ability = new EntersBattlefieldThisOrAnotherTriggeredAbility(
+                new GainControlTargetEffect(Duration.EndOfTurn), filter, true, true);
+        ability.addEffect(new UntapTargetEffect().setText("Untap that creature"));
+        ability.addEffect(new GainAbilityTargetEffect(HasteAbility.getInstance(), Duration.EndOfTurn).setText("It gains haste until end of turn."));
+        ability.addEffect(new GainAbilityTargetEffect(FlyingAbility.getInstance(), Duration.EndOfTurn).setText("It gains flying until end of turn."));
```

Comment:
First add flying ability "it gains flying", then add haste ability "and haste until end of turn".

Then add delayed trigger here. There's no line break in the card text, it's part of the same ability. I recommend not making a new common class for the damage dealing, but just a custom class for this card (as not yet used elsewhere, and the target order is unusual). You can simplify it a lot for just being used here - no constructor params, hardcode static text, second target (land) deals 3 damage to first target (creature), with both targets added to this ability only.